packages/vite/src/index.ts

```diff
@@ -128,7 +128,7 @@ export function imagetools(userOptions: Partial<VitePluginOptions> = {}): Plugin
         if (viteConfig.command === 'serve') {
           const id = await generateImageID(srcURL, config, img)
           generatedImages.set(id, image)
-          metadata.src = basePath + id
+          metadata.src = path.join(viteConfig?.server?.origin ?? viteConfig?.origin ?? '', basePath) + id;
```

Comment:
You can't specify `origin` at the top level of the Vite config, can you?
```suggestion
          metadata.src = path.join(viteConfig?.server?.origin ?? '', basePath) + id;
```

Reply:
Looks like you are right. I think I saw it on the resolved config object but it doesn't seem to be part of the types so let's not use it in any case.